packages/commonwealth/client/scripts/state/api/threads/editThreadStage.ts

```diff
@@ -0,0 +1,56 @@
+import { useMutation } from '@tanstack/react-query';
+import axios from 'axios';
+import Thread from 'models/Thread';
+import { ThreadStage } from 'models/types';
+import app from 'state';
+import { EXCEPTION_CASE_threadCountersStore } from '../../ui/thread';
+import { updateThreadInAllCaches } from './helpers/cache';
+
+interface EditThreadStageProps {
+  chainId: string;
+  threadId: number;
+  stage: ThreadStage
+}
+
+const editThreadStage = async ({
+  chainId,
+  threadId,
+  stage
+}: EditThreadStageProps) => {
+  const response = await axios.post(`${app.serverUrl()}/updateThreadStage`, {
```

Comment:
not for this PR but would be probably ideal if we refactor these kind of thread actions to be more RESTful? cc @rbennettcw

Reply:
+1

Reply:
my first pass solution is that we should unify these thread modification routes under a single PUT call, with differential permissioning based on which data is being updated -- but we should definitely ticket up further API unification.

Reply:
> my first pass solution is that we should unify these thread modification routes under a single PUT call, with differential permissioning based on which data is being updated -- but we should definitely ticket up further API unification.

I think https://github.com/hicommonwealth/commonwealth/issues/4664 was created for this. Resolving this convo